libobs-opengl/gl-nix.c

```diff
@@ -0,0 +1,127 @@
+/******************************************************************************
+    Copyright (C) 2019 by Jason Francis <cycl0ps@tuta.io>
+
+    This program is free software: you can redistribute it and/or modify
+    it under the terms of the GNU General Public License as published by
+    the Free Software Foundation, either version 2 of the License, or
+    (at your option) any later version.
+
+    This program is distributed in the hope that it will be useful,
+    but WITHOUT ANY WARRANTY; without even the implied warranty of
+    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+    GNU General Public License for more details.
+
+    You should have received a copy of the GNU General Public License
+    along with this program.  If not, see <http://www.gnu.org/licenses/>.
+******************************************************************************/
+
+#include "gl-nix.h"
+#include "gl-nix-egl-x11.h"
+#include "gl-nix-glx.h"
+
+#ifdef ENABLE_WAYLAND
+#include "gl-nix-egl-wayland.h"
+#endif
+
+static const struct gl_winsys_vtable *gl_vtable = NULL;
+
+static void init_winsys(void)
+{
+	assert(gl_vtable == NULL);
+
+	switch (obs_get_platform()) {
+	case OBS_PLATFORM_DEFAULT:
+		if (getenv("OBS_USE_EGL")) {
```

Comment:
expose all platforms as platforms not special env vars used deep within the weeds of the code.

If you want to use an env var use it where the platform is chosen at the start of the application like you do with wayland.

Reply:
Now there is a X11/EGL platform, and the environment variable is used to force it.